xds/internal/balancer/clusterresolver/configbuilder.go

```diff
@@ -98,47 +97,27 @@ func buildPriorityConfig(priorities []priorityConfig, xdsLBPolicy *internalservi
 			}
 			retConfig.Priorities = append(retConfig.Priorities, names...)
 			retAddrs = append(retAddrs, addrs...)
-			var odCfgs map[string]*outlierdetection.LBConfig
-			if envconfig.XDSOutlierDetection {
-				odCfgs = convertClusterImplMapToOutlierDetection(configs, p.mechanism.outlierDetection)
-				for n, c := range odCfgs {
-					retConfig.Children[n] = &priority.Child{
-						Config: &internalserviceconfig.BalancerConfig{Name: outlierdetection.Name, Config: c},
-						// Ignore all re-resolution from EDS children.
-						IgnoreReresolutionRequests: true,
-					}
-				}
-				continue
-			}
-			for n, c := range configs {
+			odCfgs := convertClusterImplMapToOutlierDetection(configs, p.mechanism.outlierDetection)
+			for n, c := range odCfgs {
 				retConfig.Children[n] = &priority.Child{
-					Config: &internalserviceconfig.BalancerConfig{Name: clusterimpl.Name, Config: c},
+					Config: &internalserviceconfig.BalancerConfig{Name: outlierdetection.Name, Config: c},
```

Comment:
Its not obvious to me why we are changing this to `outlierdetection.Name` here. Could you lmk

Reply:
This is how the diffs rendered.

The remaining code is the code from inside the `if envconfig.XDSOutlierDetection`, and that ends with a `continue`.  The code on the left from 113-120 was dead code if `XDSOutlierDetection` was always true.  But the diffs combined them since there were similar elements.

Reply:
oh i see. SG